server/src/controllers/eventController.ts

```diff
@@ -135,6 +138,97 @@ eventController.post('/checkout', async (req: Request, res: Response) => {
   }
 });
 
+/**
+ * @swagger
+ * /2/events/reader-intent:
+ *   post:
+ *     summary: Create Reader Payment Intent
+ *     tags:
+ *     - New Event API
+ */
+
+eventController.post('/reader-intent', async (req: Request, res: Response) => {
+  const {cartItems} = req.body;
+  let paymentIntentID = "";
+  let clientSecret = "";
+
+  try {
+    if (!cartItems.length) {
+      return res.status(400).json({error: 'Cart is empty'});
+    }
+    const {cartRows, orderItems, orderTotal, eventInstanceQueries} =
+      await getOrderItems(cartItems, prisma);
+
+    if (orderTotal > 0) {
+      const {id, secret} = await createStripePaymentIntent(
+        orderTotal * 100
+      )
+      paymentIntentID = id;
+      clientSecret = secret;
+    }
+    res.json({id: paymentIntentID, secret: clientSecret});
+  } catch (error) {
+    console.error(error);
+    if (error instanceof InvalidInputError) {
+      res.status(error.code).json(error.message);
+      return;
+    }
+    res.status(500).json(error);
+  }
+});
+
+/**
+ * @swagger
+ * /2/events/reader-checkout:
+ *   post:
+ *     summary: Request Payment for intent and fulfill order
+ *     tags:
+ *     - New Event API
+ */
+
+eventController.post('/reader-checkout', async (req: Request, res: Response) => {
+  const {cartItems, paymentIntentID, readerID, discount} = req.body;
+  let orderID = 0;
+  let contactid = 1; //database value for anonymous
```

Comment:
This should be the actual contacts `contactid` and not hard-coded

Reply:
Reid and I were under the impression that the card reader would be used to collect payment immediately before an event, so collecting contact information would be too time consuming if there is a line, hence why we just set the contact to database value for anonymous.

If this is not the case, we can have the reader checkout go to the contact info form before the reader payment processing page. Or, perhaps, whoever is operating the reader could have a checkbox that determines whether contact info is collected or not as part of the payment process so that the reader is more context flexible. Or, maybe, just an email is collected as a middle-ground, since, based on a quick look, email is how people are attached to accounts. What do you think?

Reply:
The email address is a unique identifier of a contact. Does stripe return an email address associated with the card from the reader checkout? If there is really no way to get an email other then manually entering it, then we can skip associating the order/tickets with a contact altogether

Reply:
My understanding is that there is no additional information automatically associated with a card on a card_present to a terminal reader, at least none that stripe gives you access to. If we were to not have a contactid, would that mean passing undefined to contactid in orderfulfillment?

Reply:
If the result would be that the contactid is null in the order then I think yes

Reply:
contactid is now null in terminal reader orders. Note: I had to change and migrate the database scheme to let contactid_fk be nullable. Here is what a completed reader order looks like now
<img width="823" alt="Code_msXbT7p2Ho" src="https://github.com/WonderTix/WonderTix/assets/127803000/299cb0d2-084e-4e0b-994c-e86d59355b71">

Reply:
@ben16williams for some reason the github actions is compiling with an error thats just nonexistant on my machine. On top of that, i didn't even change anything that affected this part of the code. have you ever encountered this? 
<img width="620" alt="msedge_AOs09jtS18" src="https://github.com/WonderTix/WonderTix/assets/127803000/2fcb1674-7234-443d-9059-18a9ca4cba64">

Reply:
nevermind i figured it out